### PR TITLE
test: remove warning for unused variable

### PR DESCRIPTION
**Pull request by mdionisio**

### CODE REVIEW

**Comment by dvdhrm** [comment]
We generally do not support -Wunused-parameter in c-utils, so I would rather avoid annotating them. Sadly there is not any real consensud how to mark such variables in C, so we just avoid the compiler warning alltogether. It isnt enabled by -Wall either, is it?

File: src/meson.build [hunk 1]
```diff
@@ -57,7 +57,7 @@ endif
 # target: test-*
 #
 
-test_api = executable('test-api', ['test-api.c'], link_with: libcrbtree_shared)
+test_api = executable('test-api', ['test-api.c'], dependencies: libcrbtree_dep)
```
Comment (dvdhrm) [inline src/meson.build]: We want the API tests to link dynamically so we test the symbol visibility. This change would disable that test.

**Comment by mdionisio** [comment]
if you build with warning_level=2 you see the problem. And if you use this project in another project as meson subproject. If the main program is compiled with
warning_level=2 and werror=true the result is that all not build for useless warning in test.

do you prefere a simple modification that disable -Wunused-parameter in any case building test? This can be done in meson with something similar to
`
cc = meson.get_compiler('c')
if cc.has_argument('-Wno-unused-parameter')
        add_project_arguments('-Wno-unused-parameter', language: 'c')
endif
`

or with some pragma in c test code.

But I'm not sure that -Wno-unused-parameter' works correctly with also warning as error. 

So if for you one of the fix proposed is ok I can try to make some test and send you a updated pull request.

**Comment by mdionisio** [comment]
I upgrade the merge request to only disable some warning

**Comment by dvdhrm** [comment]
We now import the standard cflags array from the c-stdaux dependency. This includes `-Wno-unused-parameter` together with a short rationale in c-stdaux.

Thanks for the report! I did eventually find the time to finally make sure the cflags are properly kept in the build instructions!

**Pull request closed without merging**